build.sbt

```diff
@@ -67,34 +60,33 @@ def isolateAllTests(tests: Seq[TestDefinition]) = tests map { test =>
   new Group(test.name, Seq(test), SubProcess(options))
 } toSeq
 
-// Subproject definitions begin
-
-// -- Rocket Chip --
-
-// This needs to stay in sync with the chisel3 and firrtl git submodules
 val chiselVersion = "3.4.1"
-lazy val chiselRef = ProjectRef(workspaceDirectory / "chisel3", "chisel")
-lazy val chiselLib = "edu.berkeley.cs" %% "chisel3" % chiselVersion
-lazy val chiselLibDeps = (chiselRef / Keys.libraryDependencies)
-// While not built from source, *must* be in sync with the chisel3 git submodule
-// Building from source requires extending sbt-sriracha or a similar plugin and
-//   keeping scalaVersion in sync with chisel3 to the minor version
-lazy val chiselPluginLib = "edu.berkeley.cs" % "chisel3-plugin" % chiselVersion cross CrossVersion.full
+
+lazy val chiselSettings = Seq(
+  libraryDependencies ++= Seq("edu.berkeley.cs" %% "chisel3" % chiselVersion),
+  addCompilerPlugin("edu.berkeley.cs" % "chisel3-plugin" % chiselVersion cross CrossVersion.full))
 
 val firrtlVersion = "1.4.1"
-lazy val firrtlRef = ProjectRef(workspaceDirectory / "firrtl", "firrtl")
-lazy val firrtlLib = "edu.berkeley.cs" %% "firrtl" % firrtlVersion
-val firrtlLibDeps = settingKey[Seq[sbt.librarymanagement.ModuleID]]("FIRRTL Library Dependencies sans antlr4")
-Global / firrtlLibDeps := {
-  // drop antlr4 compile dep. but keep antlr4-runtime dep. (compile needs the plugin to be setup)
-  (firrtlRef / Keys.libraryDependencies).value.filterNot(_.name == "antlr4")
-}
 
- // Rocket-chip dependencies (subsumes making RC a RootProject)
+lazy val firrtlSettings = Seq(libraryDependencies ++= Seq("edu.berkeley.cs" %% "firrtl" % firrtlVersion))
+
+// In some projects we override the default versions of Chisel and friends.
+// This map captures the expected defaults used by projects under Chipyard.
+lazy val chipyardMandatedVersions = Map(
+  "chisel-iotesters" -> "1.5.4",
+  "firrtl-interpreter" -> "1.4.4",
+  "treadle" -> "1.3.4",
+  "chisel3" -> chiselVersion,
+  "firrtl" -> firrtlVersion
+)
```

Comment:
Why are you using `.4` for other things but `3.4.1` and `1.4.1` for chisel3 and firrtl? I'll note that using chisel-iotesters 1.5.4 might override the chisel3 to 3.4.4 anyway, you should check this in sbt with `fullClasspath` to see what actual jars you're using.

Reply:
It is also possible that because you're giving chisel3 directly though it might pick your version.

Reply:
I just picked the versions we had submoduled in.  Looking at the class path FIRRTL 1.4.4 is showing up in some projects likely because of the first three dependencies. I can bump to chisel 3.4.4 and FIRRTL 1.4.4 but i was scared of changing too much, especially given this is going to be quickly followed by a 3.5 bump.

Reply:
Bumped to X.4.4.